<a href="https://colab.research.google.com/github/cvillegas188/LaSalleCarlosV/blob/main/EJERCICIOS/Modelos_Pre_Entrenados_CIBER.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
from tensorflow.keras.preprocessing import image

In [19]:
#VGG16
from tensorflow.keras.applications.vgg16 import VGG16, preprocess_input, decode_predictions
model = VGG16(weights='imagenet')
model_name ='VGG16'

img_path = '/content/drive/MyDrive/Inteligencia-Computacional/perfil.jpg'
img = image.load_img(img_path, target_size=(224,224))
img_array = image.img_to_array(img)
img_array_expanded_dims = np.expand_dims(img_array, axis=0)
img_preprocessed = preprocess_input(img_array_expanded_dims)

In [22]:
#ResNet50
from tensorflow.keras.applications.resnet50 import ResNet50, preprocess_input, decode_predictions
model = ResNet50(weights='imagenet')
model_name ='ResNet50'

img_path = '/content/drive/MyDrive/Inteligencia-Computacional/perfil.jpg'
img = image.load_img(img_path, target_size=(224, 224))
img_array = image.img_to_array(img)
img_array_expanded_dims = np.expand_dims(img_array, axis=0)
img_preprocessed = preprocess_input(img_array_expanded_dims)

In [25]:
#Inception V3
from tensorflow.keras.applications.inception_v3 import InceptionV3, preprocess_input, decode_predictions
model = InceptionV3(weights='imagenet')
model_name ='Inception V3'

img_path = '/content/drive/MyDrive/Inteligencia-Computacional/perfil.jpg'
img = image.load_img(img_path, target_size=(299, 299))
img_array = image.img_to_array(img)
img_array_expanded_dims = np.expand_dims(img_array, axis=0)
img_preprocessed = preprocess_input(img_array_expanded_dims)

In [28]:
#Xception
from tensorflow.keras.applications.xception import Xception, preprocess_input, decode_predictions
model = Xception(weights='imagenet')
model_name ='Xception'

img_path = '/content/drive/MyDrive/Inteligencia-Computacional/perfil.jpg'
img = image.load_img(img_path, target_size=(299, 299))
img_array = image.img_to_array(img)
img_array_expanded_dims = np.expand_dims(img_array, axis=0)
img_preprocessed = preprocess_input(img_array_expanded_dims)

In [31]:
#MobileNet
from tensorflow.keras.applications.mobilenet import MobileNet, preprocess_input, decode_predictions
model = MobileNet(weights='imagenet')
model_name ='MobileNet'

img_path = '/content/drive/MyDrive/Inteligencia-Computacional/perfil.jpg'
img = image.load_img(img_path, target_size=(224, 224))
img_array = image.img_to_array(img)
img_array_expanded_dims = np.expand_dims(img_array, axis=0)
img_preprocessed = preprocess_input(img_array_expanded_dims)

17225924/17225924 [==============================] - 0s 0us/step


In [34]:
#EfficientNetB0
from tensorflow.keras.applications.efficientnet import EfficientNetB0, preprocess_input, decode_predictions
model = EfficientNetB0(weights='imagenet')
model_name ='EfficientNetB0'

img_path = '/content/drive/MyDrive/Inteligencia-Computacional/perfil.jpg'
img = image.load_img(img_path, target_size=(224, 224))
img_array = image.img_to_array(img)
img_array_expanded_dims = np.expand_dims(img_array, axis=0)
img_preprocessed = preprocess_input(img_array_expanded_dims)

21834768/21834768 [==============================] - 0s 0us/step


In [37]:
#DenseNet121
from tensorflow.keras.applications.densenet import DenseNet121, preprocess_input, decode_predictions
model = DenseNet121(weights='imagenet')
model_name ='DenseNet121'

img_path = '/content/drive/MyDrive/Inteligencia-Computacional/perfil.jpg'
img = image.load_img(img_path, target_size=(224, 224))
img_array = image.img_to_array(img)
img_array_expanded_dims = np.expand_dims(img_array, axis=0)
img_preprocessed = preprocess_input(img_array_expanded_dims)

33188688/33188688 [==============================] - 0s 0us/step


In [38]:
predictions = model.predict(img_preprocessed)

1/1 [==============================] - 3s 3s/step


In [39]:
results = decode_predictions(predictions, top=3)[0]
print('Model: %s' % (model_name))
for result in results:
    print('%s: %.2f%%' % (result[1], result[2] * 100))

Model: DenseNet121
Chihuahua: 97.57%
toy_terrier: 0.83%
Norwich_terrier: 0.31%


# **Resultados:**

VGG16
*   0s
*   94%

ResNet50
*   1s
*   89.4%

InceptionV3
*   1s
*   94.39%

Xception
*   1s
*   88.41%

MobileNetV2
*   1s
*   98.56%

EfficientNetB0
*   3s
*   69.83%

DenseNet121
*   3s
*   97.57%